control_panel_api/tools.py

```diff
@@ -0,0 +1,66 @@
+import secrets
+
+from django.conf import settings
+from rest_framework.exceptions import APIException
+
+from control_panel_api.helm import Helm
+
+
+SUPPORTED_TOOL_NAMES = [
+    'rstudio',
+]
+
+
+class UnsupportedToolException(APIException):
+    tools_list = ", ".join(SUPPORTED_TOOL_NAMES)
+
+    status_code = 400
+    default_detail = f'Unsupported tool, tool_name must be one of: {tools_list}'
+    default_code = 'unsupported_tool'
+
+
+class Tool():
+
+    def __init__(self, name):
+        self.name = self._validate_name(name)
+        self.helm = Helm()
+
+        self.auth_client_domain = self._get_auth_client_config('domain')
+        self.auth_client_id = self._get_auth_client_config('id')
+        self.auth_client_secret = self._get_auth_client_config('secret')
+
+    def deploy_for(self, user):
+        """
+        Deploy the given tool in the user namespace.
+
+        >>> rstudio = Tool('rstudio')
+        >>> rstudio.deploy_for(alice)
+        """
+
+        username = user.username.lower()
+        auth_proxy_cookie_secret = secrets.token_hex(32)
+        tool_cookie_secret = secrets.token_hex(32)
+
+        self.helm.upgrade_release(
+            f'{username}-{self.name}',
```

Comment:
Not sure, but I think you might need to use `user.k8s_namespace` here instead of `username`, and below in `username={username}`

Reply:
I used the same naming convention used currently when deploying using the jenkins job to avoid any confusion or bugs: https://github.com/ministryofjustice/analytics-platform-control-panel/blob/jenkins-control-panel/scripts/deploy_rstudio#L31